app/channel.go

```diff
@@ -596,10 +596,90 @@ func (a *App) postChannelPrivacyMessage(user *model.User, channel *model.Channel
 	return nil
 }
 
-func (a *App) RestoreChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
-	if err := a.Srv.Store.Channel().Restore(channel.Id, model.GetMillis()); err != nil {
+func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Channel, *model.AppError) {
+	ihc := make(chan store.StoreResult, 1)
+	ohc := make(chan store.StoreResult, 1)
+
+	go func() {
+		webhooks, err := a.Srv.Store.Webhook().GetIncomingByChannel(channel.Id)
+		ihc <- store.StoreResult{Data: webhooks, Err: err}
+		close(ihc)
+	}()
+
+	go func() {
+		outgoingHooks, err := a.Srv.Store.Webhook().GetOutgoingByChannel(channel.Id, -1, -1)
+		ohc <- store.StoreResult{Data: outgoingHooks, Err: err}
+		close(ohc)
+	}()
+
+	var user *model.User
+	if userId != "" {
+		var err *model.AppError
+		user, err = a.Srv.Store.User().Get(userId)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	ihcresult := <-ihc
+	if ihcresult.Err != nil {
+		return nil, ihcresult.Err
+	}
+
+	ohcresult := <-ohc
+	if ohcresult.Err != nil {
+		return nil, ohcresult.Err
+	}
+
+	incomingHooks := ihcresult.Data.([]*model.IncomingWebhook)
+	outgoingHooks := ohcresult.Data.([]*model.OutgoingWebhook)
+
+	if channel.DeleteAt == 0 {
+		err := model.NewAppError("undeleteChannel", "api.channel.undelete_channel.undeleted.app_error", nil, "", http.StatusBadRequest)
 		return nil, err
 	}
+
+	if user != nil {
+		T := utils.GetUserTranslations(user.Locale)
+
+		post := &model.Post{
+			ChannelId: channel.Id,
+			Message:   fmt.Sprintf(T("api.channel.undelete_channel.unarchived"), user.Username),
+			Type:      model.POST_CHANNEL_UNDELETED,
```

Comment:
I'm wondering, if we really need another post type for this?

Reply:
There is `model.POST_SYSTEM_GENERIC` for generic system posts. If do specific rendering on web or mobile is needed, we can use this.

Reply:
what do you mean by specific rendering?

Reply:
The webapp and mobile support specific rendering of post types. E.g. some post type could be displayed with blue text.

Reply:
ah. yes we are rendering this system message differently in mobile

Reply:
Just want to double check why you need to render this differently on mobile? Why isn't the "normal" system post fine?

Reply:
It is actually rendered basically the same. the archive channel is doing it the same way and is basically the same functionality. so i was kind of going for consistency.  I also see the other post types like HEADER_CHANGE is rendering the same.  🤔

Reply:
Do we need it anytime we are rendering a Username?  I think its that.

Reply:
@saturninoabril Can you help with this discussion as I don't know enough about the webapp and mobile code.

Reply:
Most of the types are not strictly needed, I think is correct to keep it consistent with other types, like `POST_CHANNEL_DELETED`, the `POST_SYSTEM_GENERIC` shouldn't be an excuse to not properly define the correct types for the posts.